File: backend/app/schemas/llm_provider_config.py
"""LLM provider config API schemas (v1.2.7-B).

API key 只从 Create/Update/Test 请求进入，response 永不返回明文 key（仅
`has_api_key` 标记）。
"""

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, Field


class LlmConfigCreateRequest(BaseModel):
    provider: str = Field(min_length=1, max_length=32)
    display_name: str = Field(min_length=1, max_length=120)
    model_id: str = Field(min_length=1, max_length=160)
    base_url: str | None = Field(default=None, max_length=255)
    api_key: str | None = Field(default=None, max_length=512)
    is_active: bool = False


class LlmConfigUpdateRequest(BaseModel):
    display_name: str | None = Field(default=None, max_length=120)
    provider: str | None = Field(default=None, min_length=1, max_length=32)
    model_id: str | None = Field(default=None, min_length=1, max_length=160)
    base_url: str | None = Field(default=None, max_length=255)
    api_key: str | None = Field(default=None, max_length=512)
    is_active: bool | None = None


class LlmConfigSetActiveRequest(BaseModel):
    is_active: bool


class LlmConfigTestRequest(BaseModel):
    provider: str = Field(min_length=1, max_length=32)
    model_id: str = Field(min_length=1, max_length=160)
    base_url: str | None = Field(default=None, max_length=255)
    api_key: str | None
    # 复用已存配置（API key 未明文回传）时传 True，service 用加密 key 尝试。
    use_stored_key: bool = False


class LlmConfigResponse(BaseModel):
    id: UUID
    provider: str
    display_name: str
    model_id: str
    base_url: str | None
    has_api_key: bool
    is_active: bool
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class LlmConfigListResponse(BaseModel):
    items: list[LlmConfigResponse]
    total: int
    active_id: UUID | None = None


class LlmConfigTestResponse(BaseModel):
    ok: bool
    latency_ms: int | None = None
    message: str

File: frontend/src/api/llmProviderConfig.ts
/** 用户模型配置中心 API（v1.2.7-B：LLM provider configs 管理）。
 * 后端 /app/api/v1/routes/llm_provider_configs.py。
 */

import { apiRequest } from './client';

export type LlmProviderType = 'deepseek' | 'openai' | 'openrouter' | 'custom';

export interface LlmProviderConfigItem {
  id: string;
  provider: LlmProviderType;
  display_name: string;
  model_id: string;
  base_url: string | null;
  /** 是否已保存 API Key（永不返回明文）。 */
  has_api_key: boolean;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

export interface LlmConfigListPayload {
  items: LlmProviderConfigItem[];
  total: number;
  active_id: string | null;
}

export interface LlmConfigUpsertPayload {
  provider: LlmProviderType;
  display_name: string;
  model_id: string;
  base_url?: string | null;
  api_key?: string | null;
  is_active?: boolean;
}

export interface LlmConfigTestResult {
  ok: boolean;
  latency_ms?: number | null;
  message: string;
}

export const llmConfigKeys = {
  all: ['llm-configs'] as const,
  list: () => [...llmConfigKeys.all, 'list'] as const,
  detail: (id: string) => [...llmConfigKeys.all, 'detail', id] as const,
};

export async function listLlmConfigs(): Promise<LlmConfigListPayload> {
  return apiRequest<LlmConfigListPayload>('/llm-configs');
}

export async function createLlmConfig(
  payload: LlmConfigUpsertPayload,
): Promise<LlmProviderConfigItem> {
  return apiRequest<LlmProviderConfigItem>('/llm-configs', {
    method: 'POST',
    body: payload,
  });
}

export async function updateLlmConfig(
  configId: string,
  payload: Partial<LlmConfigUpsertPayload>,
): Promise<LlmProviderConfigItem> {
  return apiRequest<LlmProviderConfigItem>(`/llm-configs/${configId}`, {
    method: 'PUT',
    body: payload,
  });
}

export async function deleteLlmConfig(configId: string): Promise<void> {
  await apiRequest<void>(`/llm-configs/${configId}`, { method: 'DELETE' });
}

export async function setActiveLlmConfig(
  configId: string,
  active: boolean,
): Promise<LlmProviderConfigItem> {
  return apiRequest<LlmProviderConfigItem>(`/llm-configs/${configId}/active`, {
    method: 'POST',
    body: { is_active: active },
  });
}

export async function testLlmConnection(
  configId: string,
): Promise<LlmConfigTestResult> {
  return apiRequest<LlmConfigTestResult>(`/llm-configs/${configId}/test`, {
    method: 'POST',
  });
}

export async function testLlmConnectionDraft(
  payload: Partial<LlmConfigUpsertPayload>,
): Promise<LlmConfigTestResult> {
  return apiRequest<LlmConfigTestResult>('/llm-configs/test', {
    method: 'POST',
    body: payload,
  });
}
